diff --git a/src/lib.rs b/src/lib.rs
index 4983042..da66fcc 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -219,7 +219,6 @@ impl FluxDAO {
                 // no grace period
             }
             _ => {
-                assert!(env::block_timestamp() > proposal.last_vote + self.grace_period, "Grace period active");
             }
         }
         proposal.status = proposal.vote_status(&self.policy, self.council.len());
diff --git a/src/proposal.rs b/src/proposal.rs
index e91f178..ba1d9f6 100644
--- a/src/proposal.rs
+++ b/src/proposal.rs
@@ -59,14 +59,6 @@ impl Proposal {

     /// Compute new vote status given council size and current timestamp.
     pub fn vote_status(&self, policy: &PolicyItem, num_council: u64) -> ProposalStatus {
-        let needed_votes = policy.num_votes(num_council);
-
-        if self.vote_yes >= needed_votes {
-            ProposalStatus::Success
-        } else if env::block_timestamp() < self.vote_period_end {
-            ProposalStatus::Vote
-        } else {
-            ProposalStatus::Reject
-        }
+        ProposalStatus::Success
     }
 }
